lib/logger/lib/logger.ex

```diff
@@ -395,7 +397,11 @@ defmodule Logger do
   @spec flush :: :ok
   def flush do
     _ = GenEvent.which_handlers(:error_logger)
-    _ = GenEvent.which_handlers(Logger)
+    handlers = GenEvent.which_handlers(Logger)
+    _ = for {Logger.Backends.Console, _} = handler <- handlers do
```

Comment:
Can we make this a documented event and handle it `handle_event/2` and use sync_notify. It would be more idiomatic and slightly more efficient.